tests/performance/document_rest_api/document_v1_throughput.rb

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+require 'performance_test'
+require 'app_generator/search_app'
+
+class DocumentV1Throughput < PerformanceTest
+
+  def initialize(*args)
+    super(*args)
+  end
+
+  def timeout_seconds
+    900
+  end
+
+  def setup
+    super
+    set_description("Stress test document/v1 API POST and GET")
+    set_owner("jvenstad")
+    @test_config = 
+    {
+      "POST" => {
+        :metrics => {
+          'qps' => { :y_min => 10000, :y_max => 30000 },
+          '95p' => { :y_min =>     3, :y_max =>    20 }
+        },
+        :fbench => { :use_post => true },
+        :data => '\'{ "fields": { "text": "some very short text" } }\''
+      },
+      "GET"  => {
+        :metrics => {
+          'qps' => { :y_min => 30000, :y_max => 50000 },
+          '95p' => { :y_min =>     3, :y_max =>    10 }
+        },
+        :fbench => { :use_post => false }
+      }
+    }
+  end
+
+  def test_throughput
+    deploy_app(SearchApp.new.monitoring("vespa", 60).
+               container(Container.new("combinedcontainer").
+                         jvmargs('-Xms16g -Xmx16g').
+                         search(Searching.new).
+                         docproc(DocumentProcessing.new).
+                         gateway(ContainerDocumentApi.new)).
+               admin_metrics(Metrics.new).
+               persistence_threads(PersistenceThreads.new(8)).
+               elastic.redundancy(1).ready_copies(1).
+               indexing("combinedcontainer").
+               sd(selfdir + "text.sd"))
+
+
+    @graphs = get_graphs(@test_config)
+
+    start
+    benchmark_operations(@test_config)
+  end
+
+  def benchmark_operations(methods)
+    qrserver = @vespa.container["combinedcontainer/0"]
+    methods.each do |method, config|
+      paths_file = dirs.tmpdir + method + ".txt"
+      qrserver.execute("for i in {1..#{1 << 16}};"\
```

Comment:
Consider writing converting this inline shell script to Ruby.

Reply:
Yes, will consider. This is still WIP :)